### DISCUSSION TITLE
Override not working

### DISCUSSION THREAD

**Discussion by Nixinova** [github-linguist/linguist#5294] [Classification] [answered]
My repo's [languages](https://github.com/Nixinova/Wikity/search?l=javascript) shows [a `.ts` file](https://github.com/Nixinova/Wikity/blob/5f875c4609d00887b39668b8190c8f023a967de0/src/cli.ts) as JS even though I have explicitly used [an override](https://github.com/Nixinova/Wikity/blob/5f875c4609d00887b39668b8190c8f023a967de0/.gitattributes#L8) because of #5251.

```gitignore
*.ts linguist-language=TypeScript
```

Caching wouldn't be an issue because this repo only has one single commit.

What would the issue here be?

**Comment by Alhadis** [comment, marked as answer]
> What would the issue here be?

Pushing a change to `.gitattributes` will only impact classification once *another* change gets pushed, which forces GitHub to refresh its now-stale cache. When I forked your repository and pushed a [minor change](https://github.com/Cutlery-Drawer/Wikity/commit/15fb7c602ed04bc21ce29a7e8c74dadcf54ac177), I saw that `cli.ts` *was* being identified as TypeScript:

<img src="https://user-images.githubusercontent.com/2346707/112740832-de8f5300-8fcb-11eb-9525-4a72de4e362b.png" width="925" alt="Figure 1" />

Though I'm not sure what's going on with the sidebar…

**Reply by Nixinova** [reply]
> once another change gets pushed

Ah, alright.

> now-stale cache

It was never cached in the first place, though.

**Reply by Alhadis** [reply]
> It was never cached in the first place, though.

Since it's a background job queued in an already resource-intensive pipeline, I'd say it's deliberate—it's more efficient to pick `.gitattributes` rules up whilst you're scanning the repository the first time, rather than scanning every directory once for new overrides, then scanning them all again to compute language-stats.

That's just a blind guess though, as I've no knowledge of GitHub's actual internals.

**Reply by Nixinova** [reply]
Hm, Ive pushed two commits and the file has been reindexed but its [still showing up](https://github.com/Nixinova/Wikity/search?l=javascript) as JS.

**Reply by Alhadis** [reply]
If you pop open the devtools, you'll see the same classes as those in my earlier screenshot. IIRC, there's some dissonance going on at the moment between Linguist and GitHub's indexing engine. I can never seem to remember which parts Linguist is responsible for, and which to blame on somebody else. 😂

**Reply by lildude** [reply]
> Hm, Ive pushed two commits and the file has been reindexed but its [still showing up](https://github.com/Nixinova/Wikity/search?l=javascript) as JS.

You've encountered this before 😉 This is the problem I described right at the beginning of my response at https://github.com/github/linguist/discussions/5249#discussioncomment-428558
